### ISSUE TITLE
WPF project causing .deps.json file to be lost from PublishItemsOutputGroup

### ISSUE THREAD

**Issue by wasabii** [dotnet/sdk#16995]
Unsure this is the right forum for this. But, maybe? Seems right.

I have a bunch of custom installer-generating stuff that is invoking PublishItemsOutputGroup on various projects to collect the proper publish output (post trimming, etc). Introducing a WPF application into the mix and I noticed that the .deps.json file isn't appearing in the output of PublishItemsOutputGroup.

It does appear for .Web and .Worker SDK projects.

It DOES appear when publishing the WPF application to a directory. It looks like it's creating the file properly, it's just not getting added into the output group correctly for consuming from other MSBuild scripts.

The WPF project is using UseWPF, with SDK string "Microsoft.Net.Sdk". I don't think UseSdk is the culprit, as I tried to remove it, and got the same results.

Adding `<GenerateDependencyFile>true</>` and `<_UseBuildDependencyFile>true</>` to the `PropertyGroup` in my project file resolves the issue. So, during the PublishItemsOutputGroup target, I suspect __UseBuildDependencyFile isn't getting computed properly.

**Comment by wasabii** [comment]
Of course, I can't actually use _UseBuildDependencyFiles, since that copies a broken file. Apparently I do need the publish one. And that's just getting left out for some unknown reason still. Will keep digging.

**Comment by wasabii** [comment]
Think I got it. PublishItemsOutputGroup doesn't depend on GeneratePublishDependencyFile in anyway. It never requires ComputeFilesToPublish. But the Worker and Web SDKs sorta change that up, resulting in it working.

**Comment by sfoslund** [comment]
@wasabii it sounds like you found a solution to your issue, can this be closed?

**Comment by wasabii** [comment]
It's still a bug on the SDK as far as I can tell. You guys fix those, right?
